packages/controller-utils/src/util.ts

```diff
@@ -16,6 +16,19 @@ import { MAX_SAFE_CHAIN_ID } from './constants';
 
 const TIMEOUT_ERROR = new Error('timeout');
 
+const PROTOTYPE_POLLUTION_BLOCKLIST = ['__proto__', 'constructor'] as const;
+
+/**
+ * Checks whether a dynamic string used as an object property key
+ * could be used in a prototype pollution attack.
+ *
+ * @param key - The dynamic key to check for safety.
+ * @returns Whether the given dyanmic key is safe to use.
+ */
+export function isSafeDynamicKey(key: string): boolean {
+  return PROTOTYPE_POLLUTION_BLOCKLIST.every((badInput) => key !== badInput);
+}
```

Comment:
A few things:

- When people talk about a prototype pollution attack, do people call the offending items "keys" or "properties"? Some googling reveals that people use the latter, so to prevent confusion, do we want to use "property"?

- I'm confused by the use of "dynamic string" here. I know that prototype pollution refers to `__proto__` or `constructor` getting set to something that changes the behavior of the object, and I guess that makes it dynamic, but it's the property that's dynamic, not the name of the property.

- The description that GitHub gives if you expand the part about "

Finally I'm curious if the use of `includes` may be more readable.

Putting this together, what about:

``` suggestion
/**
 * Checks whether a dynamic string used as an object property
 * could be used in a prototype pollution attack.
 *
 * @param key - The dynamic key to check for safety.
 * @returns Whether the given dyanmic key is safe to use.
 */
export function isSafeDynamicProperty(propertyName: string): boolean {
  const PROTOTYPE_POLLUTION_BLOCKLIST
  return PROTOTYPE_POLLUTION_BLOCKLIST.every((badInput) => key !== badInput);
}
```